org-roam-doctor.el

```diff
@@ -0,0 +1,220 @@
+        (save-excursion
+          (let ((label (if (match-end 2)
+                           (match-string-no-properties 2)
+                         (org-link-unescape (match-string-no-properties 1)))))
+            (delete-region (match-beginning 0) (match-end 0))
+            (org-roam-insert nil nil label)))
+      (quit (progn
+              (replace-buffer-contents orig)
+              (goto-char p))))))
+
+(defun org-roam-doctor--remove-link ()
+  "Unlink the text at point."
+  (unless (org-in-regexp org-link-bracket-re 1)
+    (user-error "No link at point"))
+  (save-excursion
+    (let ((label (if (match-end 2)
+                     (match-string-no-properties 2)
+                   (org-link-unescape (match-string-no-properties 1)))))
+      (delete-region (match-beginning 0) (match-end 0))
+      (insert label))))
+
+(defun org-roam-doctor--resolve (msg checker)
+  "Resolve an error.
+MSG is the error that was found, which is displayed in a help buffer.
+CHECKER is a org-roam-doctor checker instance."
+  (let ((actions (org-roam-doctor-checker-actions checker))
+        c)
+    (push '("e" . ("Edit" . org-roam-doctor--recursive-edit)) actions)
+    (push '("s" . ("Skip" . org-roam-doctor--skip)) actions)
+    (with-output-to-temp-buffer "*Org-roam-doctor Help*"
+      (mapc #'princ
+            (list "Error message:\n   " msg "\n\n"))
+      (dolist (action actions)
+        (princ (format "[%s]: %s\n"
+                       (car action)
+                       (cadr action))))
+      (princ "\n\n"))
+    (shrink-window-if-larger-than-buffer
+     (get-buffer-window "*Org-roam-doctor Help*"))
+    (message "Press key for command:")
+    (unwind-protect
+        (progn
+          (cl-loop
+           do (setq c (char-to-string (read-char-exclusive)))
+           until (assoc c actions)
+           do (message "Please enter a valid key for command:"))
+          (funcall (cddr (assoc c actions)))
+          (redisplay))
+      (when (get-buffer-window "*Org-roam-doctor Help*")
+        (delete-window (get-buffer-window "*Org-roam-doctor Help*"))
+        (kill-buffer "*Org-roam-doctor Help*")))))
+
+;;;###autoload
+(defun org-roam-doctor (&optional this-buffer)
+  "Perform a check on Org-roam files to ensure cleanliness.
+If THIS-BUFFER, run the check only for the current buffer."
+  (interactive "P")
+  (let ((existing-buffers (org-roam--get-roam-buffers))
+        files)
+    (if (not this-buffer)
+        (setq files (org-roam--list-all-files))
+      (unless (org-roam--org-roam-file-p)
+        (user-error "Not in an org-roam file"))
+      (setq files (list (buffer-file-name))))
+    (save-window-excursion
+      (dolist (f files)
+        (let ((buf (find-file-noselect f)))
+          (with-current-buffer buf
+            (org-roam-doctor--check buf org-roam-doctor--checkers))
+          (unless (memq buf existing-buffers)
+            (save-buffer buf)
+            (kill-buffer buf))))))
+  (message "Linting completed."))
```

Comment:
Perhaps we could introduce a customizable variable, `org-roam-doctor-files` which defaults to `(org-roam-llist-all-files)`. `org-roam-doctor` would run its checks against the files in this list. Users/commands could then customize that list to check a subset.
e.g.
```emacs-lisp
(let ((org-roam-doctor-files `("example.org" "etc..."))) 
  (org-roam-doctor))
```
 similar to `org-agenda-files`.